### PR TITLE
8152 - Fixed the Agency V2 Page Total Obligation Over Time Tooltip 

### PR DESCRIPTION
**Pull request by anjenkin**
**High level description:**

For browsers 13" and smaller, fixed the agency v2 page Total Obligation over Time tooltip to always be visible on the page.  

**Technical details:**

Move the tooltipwrapper to wrap the whole svg instead of the element within the svg.  Tweaked scss to re-position tooltip and address design feedback.  https://data-transparency-bfs.slack.com/archives/C01JGUV1N4Q/p1641330070250000

**JIRA Ticket:**
[DEV-8152](https://federal-spending-transparency.atlassian.net/browse/DEV-8152)

**Mockup:**

See the above Slack thread for style updated made since the original mocks were developed

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [n/a] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [n/a] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [n/a] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [ ] Code review complete

### CODE REVIEW

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ const AgencyBudgetLine = ({
                 }
             );
         }
-    }, [xScale, yScale, showTodayLineAndText]);
+    }, [xScale, yScale, showTodayLineAndText, agencyBudget, height, padding.bottom, padding.left, todaysDate, width]);
```
Comment (brett-varney) [inline src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx:57]: Was this intentional or automatically added by a linter? I've had these auto-fixes really mess things up...

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx [hunk 2]
```diff
@@ -63,38 +63,26 @@ const AgencyBudgetLine = ({
...
-            </div>
-            <div className="tooltip__text">
-                <div className="budgetary-resources-tooltip__desc">Unobligated Balance</div>
-                <div className="budgetary-resources-tooltip__desc_percent">Percent of Total</div>
-                <div className="budgetary-resources-tooltip__amount">{formatMoney(rectangleData.balance)}</div>
-                <div className="budgetary-resources-tooltip__amount_percent">{rectangleData.percentOfTotal}</div>
-            </div>
-        </div>
-    );
+    }, [xScale, yScale, showTodayLineAndText, data, padding.left, padding.bottom, padding.right, padding.top, height, agencyBudget, todaysDate, width]);
```
Comment (brett-varney) [inline src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx:70]: same question

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/TotalObligationsOverTimeVisualization.jsx [hunk 3]
```diff
@@ -137,63 +140,98 @@ const TotalObligationsOverTimeVisualization = ({
         }
     }, [xDomain, todaysDate]);
 
+    const toggleTooltipVisibility = (isVisible) => {
+        setTooltipIsVisible(isVisible);
+    };
+
+    const balance = agencyBudget - data[data.length - 1].obligated;
+    const percentOfTotal = calculatePercentage(agencyBudget - data[data.length - 1].obligated, agencyBudget);
```
Comment (brett-varney) [inline src/js/components/agencyV2/visualizations/totalObligationsOverTime/TotalObligationsOverTimeVisualization.jsx:148]: Is it guaranteed that the final obligation value in the array is always the largest?

**Review by brett-varney — COMMENTED** [review]
just a couple of questions so far...

**Comment by brett-varney** [comment]
The issue still happens between 992 and about 1215px (starts overlapping the grey margin, then off the right of the page). This may not be enough to prevent merging.

image

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx (hunk 1 above)
Comment (anjenkin) [reply src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx:57]: This was the linter but I reviewed it. It make sense to add and doesn't appear to mess anything up.

File: src/js/components/agencyV2/visualizations/totalObligationsOverTime/TotalObligationsOverTimeVisualization.jsx (hunk 3 above)
Comment (anjenkin) [reply src/js/components/agencyV2/visualizations/totalObligationsOverTime/TotalObligationsOverTimeVisualization.jsx:148]: No, it's not necessarily largest obligation but its the last reported obligation (which it should be).  This array is sorted by period. 
Screen Shot 2022-01-05 at 11 40 47 AM

**Review by brett-varney — APPROVED** [review]

**Pull request merged**